.github/workflows/windows_comp.yml

```diff
@@ -0,0 +1,71 @@
+# Copyright 2022 The OpenSSL Project Authors. All Rights Reserved.
+#
+# Licensed under the Apache License 2.0 (the "License").  You may not use
+# this file except in compliance with the License.  You can obtain a copy
+# in the file LICENSE in the source distribution or at
+# https://www.openssl.org/source/license.html
+
+name: Windows Compression GitHub CI
+
+on:
+  pull_request:
+    paths:
+      - 'crypto/comp/*.c'
+  push:
+    paths:
+      - '**.c'
```

Comment:
This should just run these tests on PRs containing code in the `crypto/comp` directory, and on changes to the source code on a push; so it will avoid doing this on, e.g., documentation changes.